File: Bhatkanti/Models/Guide_Places.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Bhatkanti.Models
{
    [Table("guide_places")]
    public class Guide_Places
    {
        [Key]
        public int GP_ID { get; set; }

        [ForeignKey("Guide")]
        public int? Guide_ID { get; set; }

        [ForeignKey("Place")]
        public int? Place_ID { get; set; }

        [Required]
        public decimal Guide_Fee { get; set; }

        // Navigation Properties
        public Guide Guide { get; set; }
        public Place Place { get; set; }
    }
}

File: Backend SpringBoot/src/main/java/com/explorer/Models/GuidePlaces.java
package com.explorer.Models;

import jakarta.persistence.*;
import java.math.BigDecimal;


@Entity
@Table(name = "guide_places")
public class GuidePlaces {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int gpId;

    @ManyToOne
    @JoinColumn(name = "guide_id")
    private Guide guide;

    @ManyToOne
    @JoinColumn(name = "place_id")
    private Place place;

    @Column(nullable = true)
    private BigDecimal guideFee;

	public GuidePlaces(int gpId, Guide guide, Place place, BigDecimal guideFee) {
		super();
		this.gpId = gpId;
		this.guide = guide;
		this.place = place;
		this.guideFee = guideFee;
	}

	
	public GuidePlaces() {
		
	}
	
	public int getGpId() {
		return gpId;
	}

	public void setGpId(int gpId) {
		this.gpId = gpId;
	}

	public Guide getGuide() {
		return guide;
	}

	public void setGuide(Guide guide) {
		this.guide = guide;
	}

	public Place getPlace() {
		return place;
	}

	public void setPlace(Place place) {
		this.place = place;
	}

	public BigDecimal getGuideFee() {
		return guideFee;
	}

	public void setGuideFee(BigDecimal guideFee) {
		this.guideFee = guideFee;
	}

    
	
}

